### PR TITLE
refactor: The backend server of GUI

### LINKED ISSUE #530: Support for env overrides of configuration.toml
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]
edgex-ui

### Description [**REQUIRED**]
Allow the user to specify environment variables that override values set inside configuration.toml. This will allow users to deploy UI using custom settings easier than volume mounting a whole new configuration.toml.

### Describe the solution you'd like
Env overrides work in the same manner as other EdgeX services.

### Describe alternatives you've considered
volume mounting a whole new configuration.toml

### LINKED ISSUE #541: Replace use of gopkg.in/mgo.v2/bson with github.com/google/uuid
Code [here ](https://github.com/edgexfoundry/edgex-ui-go/blob/49cc9d365f5a5b30fcf0107d6b34acf483ef27cd/internal/domain/user.go#L21) uses gopkg.in/mgo.v2/bson for User Id.

The rest of Edgex and other portions of the UI are using github.com/google/uuid for IDs.

This code should also use github.com/google/uuid  to remove dependency on gopkg.in/mgo.v2/bson

### PR DESCRIPTION
**Pull request by badboy-huaqiao**
Close #530 , Close #541 

**1. Using go-mod-bootstrap instead of custom implementation of GUI backend server.
2. Supporting for env overrides of configuration.toml.
3. Removing all dead and unused code.
4. Update Docker file and Make file.**

Signed-off-by: Huaqiao Zhang <huaqiaoz@vmware.com>

If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-ui-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Review by bnevis-i — DISMISSED** [review]

**Review by bnevis-i — COMMENTED** [review]
@badboy-huaqiao "make test" is failing.

**Comment by badboy-huaqiao** [comment]
> @badboy-huaqiao "make test" is failing.

yes, I just updated with `-tags "no_messagebus"`

**Comment by badboy-huaqiao** [comment]
@bnevis-i em..., the first time failing is because of zmq, so I added `-tags "no_messagebus"`, that passed on my local side when running  make test-go,  but now, I don't konw what happened?

File: Makefile [hunk 1]
```diff
@@ -19,25 +29,25 @@ DOCKERS=docker_edgex_ui_go
 
 VERSION=$(shell cat ./VERSION 2>/dev/null || echo 0.0.0)
 
-GOFLAGS=-ldflags "-X github.com/edgexfoundry/edgex-ui-go.Version=$(VERSION)"
+GOFLAGS=-ldflags "-X github.com/edgexfoundry/edgex-ui-go.Version=$(VERSION)" -trimpath -mod=readonly
```
Comment (MonicaisHer) [inline Makefile:32]: @badboy-huaqiao Maybe the make build error is caused by `-mod=readonly`, which prevents `go mod tidy`? Please see [logs from the snap testing build](https://github.com/edgexfoundry/edgex-ui-go/actions/runs/3152765332/jobs/5128401564#step:2:1688)

Comment (badboy-huaqiao) [reply Makefile:32]: great 👍, I try again

**Comment by badboy-huaqiao** [comment]
@MonicaisHer em... I just saw that `make test` didn't use `-mod=readonly`, it looks like not that reason

File: Makefile (hunk 1 above)
Comment (MonicaisHer) [reply Makefile]: Did you forget to run `go mod tidy` locally before committing?

Comment (badboy-huaqiao) [reply Makefile]: I have fully tested it on my local side,  do you mean running `go mod tidy`  again to update  `go.sum` file?

Comment (badboy-huaqiao) [reply Makefile]: yeah, I run the `go mod tidy`  again, and it indeed updates the go.sum file again on my local side,  let me try again.

Comment (badboy-huaqiao) [reply Makefile]: still not work, I saw that there is  a warning on my local side when running `make test-go`, it shows 

> An attribution for github.com/edgexfoundry/go-mod-bootstrap/v2 is missing from Attribution.txt, please add
will it cause failing? but that didn't  show it's an error on my local side.

**Comment by bnevis-i** [comment]
When I run go mod tidy, using go 1.18, it adds 50 lines of indirect depedencies to go.mod.

**Comment by bnevis-i** [comment]
You can pull issue-#530 from my fork (bnevis-i) with the updated go.mod and go.sum.

**Comment by badboy-huaqiao** [comment]
> You can pull issue-#530 from my fork (bnevis-i) with the updated go.mod and go.sum.

ok, thanks, let me try again, hold on

**Comment by badboy-huaqiao** [comment]
great, it works now,  that's weird that go mod file’s lack of indirect dependencies can cause errors, I should upgrade golang to v18.x, my local version now is v1.16.x, thanks all @bnevis-i @MonicaisHer @ernestojeda , we can close it now if you think there is nothing else to do,  good night!

**Comment by bnevis-i** [comment]
Please read up on lazy loading and module graph pruning.
https://go.dev/ref/mod#lazy-loading

You are definitely overdue on go upgrade.  I'm surprised you got as far as you did.

**Comment by badboy-huaqiao** [comment]
> Please read up on lazy loading and module graph pruning. https://go.dev/ref/mod#lazy-loading
> 
> You are definitely overdue on go upgrade. I'm surprised you got as far as you did.

😊，great thanks, will do, most time is spent on front-end page works, that's weird the golang code still works on my local side.

**Comment by MonicaisHer** [comment]
@badboy-huaqiao Could you please fetch the latest main? Some snap tests fail due to missing snap-related files which have only been added to the main branch recently by this [PR](https://github.com/edgexfoundry/edgex-ui-go/pull/553).

**Comment by badboy-huaqiao** [comment]
@MonicaisHer done but it still failed, can you see that what happened?

**Review by MonicaisHer — APPROVED** [review]
@badboy-huaqiao Thank you for the changes, looks good! 
I have tested the snap packaging of this PR with config options, and it worked as expected:
```
# Install
$ snap install edgex-ui_2.2.1-dev.5_amd64.snap --dangerous
# Override env var by config options
$ snap set edgex-ui config.edgex-security-secret-store=false
$ snap set edgex-ui config.service-port=5000
$ snap start edgex-ui
```
```
# Validate env var has been overridden
$ snap logs -f edgex-ui
level=INFO app=edgex-ui-service source=httpserver.go:132 msg="Web server starting (localhost:5000)"
```
edgex-ui can be accessed through [http://localhost:5000](http://localhost:5000)

**Comment by badboy-huaqiao** [comment]
@MonicaisHer good job👍，we can close it now
